sklearn/cluster/_kmeans.py

```diff
@@ -1782,27 +1824,35 @@ def partial_fit(self, X, y=None, sample_weight=None):
         -------
         self
         """
+        is_first_call_to_partial_fit = not hasattr(self, 'cluster_centers_')
 
-        X = check_array(X, accept_sparse="csr", order="C",
-                        dtype=[np.float64, np.float32])
-        n_samples, n_features = X.shape
-        if hasattr(self.init, '__array__'):
-            self.init = np.ascontiguousarray(self.init, dtype=X.dtype)
+        X = self._validate_data(X, accept_sparse='csr',
+                                dtype=[np.float64, np.float32],
+                                order='C', accept_large_sparse=False,
+                                reset=is_first_call_to_partial_fit)
 
-        if n_samples == 0:
+        if X.shape[0] == 0:
             return self
```

Comment:
codecov reports that this is not covered

Reply:
Let me just remove this. A 0 sample array is catch by check_array before. This condition can never be achieved.